aggregator_core/src/datastore.rs

```diff
@@ -710,6 +710,34 @@ impl<C: Clock> Transaction<'_, C> {
         Ok(())
     }
 
+    /// Sets or unsets the expiration date of a task.
+    #[tracing::instrument(skip(self))]
+    pub async fn update_task_expiration(
+        &self,
+        task_id: &TaskId,
+        task_expiration: Option<&Time>,
+    ) -> Result<(), Error> {
+        let stmt = self
+            .prepare_cached(
+                "UPDATE tasks SET task_expiration = $1, updated_by = $2 WHERE task_id = $3",
+            )
+            .await?;
+        check_single_row_mutation(
+            self.execute(
+                &stmt,
+                &[
+                    /* task_expiration */
+                    &task_expiration.map(Time::as_naive_date_time).transpose()?,
+                    // TODO(#2819): set updated_at
```

Comment:
https://github.com/divviup/janus/issues/2819. I suspect omitting `updated_at` from `tasks` was intentional, as we'd assume it would always be immutable. I will change this in a separate PR, since for backporting we'll need a database migration.